eox_core/edxapp_wrapper/backends/enrollment_h_v1.py

```diff
@@ -8,21 +8,27 @@
 import logging
 from rest_framework.exceptions import APIException
 from django.contrib.auth.models import User
+from django.contrib.sites.models import Site
+from django.db import models
 from enrollment import api
 from enrollment.errors import CourseModeNotFoundError
 from enrollment.errors import CourseEnrollmentExistsError
 from course_modes.models import CourseMode
 from opaque_keys.edx.keys import CourseKey
 from student.models import CourseEnrollment
+from edxfuture import get_program
+import pprint
 
 LOG = logging.getLogger(__name__)
 
-
 def create_enrollment(*args, **kwargs):
     """
     backend function to create enrollment
     """
     errors = []
+    program_uuid = kwargs.get('bundle_id') or kwargs.get('program_id')
+    if program_uuid:
```

Comment:
this if condition here is weird. It is the classic example of a function that was extended for something not meant for it.
(which is what we are doing anyways)

What if we have `create_enrollment` be the general function, and according to the existance of `bundle_id` or `course_id` we call either `enroll_on_course` or `enroll_on_program`

Reply:
The result would naturally be the same for the caller of the API, but reading this code would be a lot cleaner

Reply:
Sure